test/view/broadcast/broadcast_game_screen_test.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:http/testing.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/network/http.dart';
+import 'package:lichess_mobile/src/network/socket.dart';
+import 'package:lichess_mobile/src/view/broadcast/broadcast_game_screen.dart';
+import 'package:lichess_mobile/src/view/broadcast/broadcast_player_widget.dart';
+
+import '../../network/fake_websocket_channel.dart';
+import '../../test_helpers.dart';
+import '../../test_provider_scope.dart';
+
+final client = MockClient((request) {
+  if (request.url.path == '/api/broadcast/-/-/$_roundId') {
+    return mockResponse(
+      _roundResponse,
+      200,
+      headers: {'content-type': 'application/json; charset=utf-8'},
+    );
+  }
+  if (request.url.path == '/api/study/$_roundId/$_gameId.pgn') {
+    return mockResponse(_gameResponse, 200, headers: {'content-type': 'application/x-chess-pgn'});
+  }
+  return mockResponse('', 404);
+});
+
+void main() {
+  group('Broadcast game screen', () {
+    testWidgets('Displays broadcast game screen', variant: kPlatformVariant, (tester) async {
+      final fakeSocket = FakeWebSocketChannel();
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const BroadcastGameScreen(
+          tournamentId: _tournamentId,
+          roundId: _roundId,
+          gameId: _gameId,
+        ),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+          webSocketChannelFactoryProvider.overrideWith((ref) {
+            return FakeWebSocketChannelFactory((_) => fakeSocket);
+          }),
+        ],
+      );
+
+      await tester.pumpWidget(app);
+
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+      await tester.pump();
+
+      expect(find.byType(Chessboard), findsOneWidget);
+      expect(find.byType(BroadcastPlayerWidget), findsNWidgets(2));
+      expect(find.text('Dastan, Muhammed Batuhan'), findsOne);
+      expect(find.text('Gokerkan, Cem Kaan'), findsOne);
+    });
+
+    testWidgets('Receives a new move of the game', variant: kPlatformVariant, (tester) async {
+      final fakeSocket = FakeWebSocketChannel();
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const BroadcastGameScreen(
+          tournamentId: _tournamentId,
+          roundId: _roundId,
+          gameId: _gameId,
+        ),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+          webSocketChannelFactoryProvider.overrideWith((ref) {
+            return FakeWebSocketChannelFactory((_) => fakeSocket);
+          }),
+        ],
+      );
+
+      await tester.pumpWidget(app);
+
+      await tester.pump();
+      // await tester.pump(); It might be interesting to know why it breaks the test
+
+      expect(find.byKey(const Key('c1-whitebishop')), findsOneWidget);
+
+      await addSocketMessage(
```

Comment:
remove the helper, and change to:
```dart
fakeSocket.addIncomingMessage(...);

await pumpEventQueue();
```

Notice the `pumpEventQueue()` this is how we ensure the event is delivered to the listener.

Reply:
This doesn't work because tests use a different environment for async. 

It needs to be wrapped in `runAsync` but even with this it doesn't work.

Reply:
If you replace this function: https://github.com/lichess-org/mobile/blob/84abbae0427e733921fec17d06e8c89089a7c17c/test/network/fake_websocket_channel.dart#L87

with:

```dart
  /// Simulates incoming messages from the server.
  Future<void> addIncomingMessages(Iterable<dynamic> messages) async {
    await _incomingController.addStream(Stream<dynamic>.fromIterable(messages));
  }
```

and then call

```dart
// await for event to be delivered
await tester.runAsync(() => await fakeSocket.addIncomingMessage(...));

// wait for controller to process event and rebuild UI
await tester.pump()
```

does it work?

Reply:
In theory we shouldn't use `runAsync` but I'm not sure how to solve this properly actually.